### PR TITLE
ci(docsearch): run algolia docsearch on push to main

### PR DESCRIPTION
**Pull request by roerohan**
## Description

Run algolia docsearch on GitHub actions.

## Resolved issues

NA

### Before submitting the PR, please take the following into consideration
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. If you don't have an issue, please create one.
- [x] Prefix your PR title with `feat: `, `fix: `, `chore: `, `docs:`, or `refactor:`.
- [x] The description should clearly illustrate what problems it solves.
- [x] Ensure that the commit messages follow our guidelines.
- [x] Resolve merge conflicts (if any).
- [x] Make sure that the current branch is upto date with the `main` branch.

### CODE REVIEW

File: .github/workflows/docsearch.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - run: |
+        echo "APPLICATION_ID=$APPLICATION_ID" >> .env
+        echo "API_KEY=$API_KEY" >> .env
+        chmod +x ./bin/crawl.sh
+        ./bin/crawl.sh
+      env:
```
Comment (vaibhavshn) [inline .github/workflows/docsearch.yml:18]: I think you should put env above run?

Similar to this:
image

Comment (roerohan) [reply .github/workflows/docsearch.yml:18]: It doesn't matter in `yml`.
It's all `key: value`

Comment (vaibhavshn) [reply .github/workflows/docsearch.yml:18]: yeah, but the syntax highlighting doesn't make it clear.

**Review by vaibhavshn — APPROVED** [review]
LGTM

**Pull request merged**